src/sinks/redis.rs

```diff
@@ -85,20 +113,42 @@ impl SinkBatchSettings for RedisDefaultBatchSettings {
     const TIMEOUT_SECS: f64 = 1.0;
 }
 
-#[derive(Clone, Debug, Deserialize, Serialize)]
+/// Configuration for the `redis`gsink.
+#[configurable_component(sink)]
+#[derive(Clone, Debug)]
 #[serde(deny_unknown_fields)]
 pub struct RedisSinkConfig {
+    #[configurable(derived)]
     encoding: EncodingConfig,
+
+    #[configurable(derived)]
     #[serde(default)]
     data_type: DataTypeConfig,
+
+    #[configurable(derived)]
     #[serde(alias = "list")]
     list_option: Option<ListOption>,
+
+    /// The Redis URL to connect to.
+    ///
+    /// The URL _must_ take the form of `protocol://server:port/db` where the protocol can either be
+    /// `redis` or `rediss` for connections secured via TLS.
     url: String,
-    key: String,
+
+    /// The Redis key to publish messages to.
+    #[configurable(metadata(templateable))]
+    #[configurable(validation(length(min = 1)))]
+    key: Template,
```

Comment:
Made this `Template` because it got parsed as a `Template` during build, and we can discover any errors with the input string way earlier up the stack... and eventually, we'd need to do this anyways when we want to encode the `Template` allowable format, etc, into the configuration schema.